### PR TITLE
👷 Melos

### LINKED ISSUE #1645: [Architecture] Reconstructing the repo with melos or other standardized monorepos
### Request Statement

The current structure of the repo is not following any practice of monorepos, we might consider reconstructing it using melos or other standardized structures.

### PR DESCRIPTION
**Pull request by AlexV525**
Resolve #1645.

### New Pull Request Checklist

- [x] I have read the [Documentation](https://pub.dev/documentation/dio/latest/)
- [x] I have searched for a similar pull request in the [project](https://github.com/cfug/dio/pulls) and found none
- [x] I have updated this branch with the latest `main` branch to avoid conflicts (via merge from master or rebase)
- [ ] I have added the required tests to prove the fix/feature I'm adding
- [ ] I have updated the documentation (if necessary)
- [x] I have run the tests without failures
- [ ] I have updated the `CHANGELOG.md` in the corresponding package

### CODE REVIEW

File: melos.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+name: dio
+repository: https://github.com/cfug/dio
+
```
Comment (kuhnroyal) [inline melos.yaml:3]: Should this have a `publish_to: none`?

Comment (AlexV525) [reply melos.yaml:3]: Does Melos contain the property? Or you're indicating the root `pubspec.yaml`?

Comment (kuhnroyal) [reply melos.yaml:3]: Ah yea, was thinking about the root pubspec - so we don't accidentally publish the `dio_workspace`.

**Comment by AlexV525** [comment]
It seems there is no need for explicit scripts at present since our workflows have already done that.

**Review by ueman — APPROVED** [review]

**Pull request merged**